### PR TITLE
Add 2023 info to past conference and proceedings pages

### LINKED ISSUE #47: add 2023 to past conferences
It belongs in the list here https://conference.scipy.org/past.html

### PR DESCRIPTION
**Pull request by achabotl**
Plus fix standardize date formatting on proceedings page.

Fixes #47.

### CODE REVIEW

**Comment by achabotl** [comment]
@matthewfeickert, Would you be willing to take a quick look at this since you seem to have some bandwidth? Thanks.

**Review by matthewfeickert — APPROVED** [review]
LGTM. Thanks @achabotl

File: content/pages/proceedings.md [hunk 1]
```diff
@@ -5,54 +5,57 @@ Save_as: proceedings/index.html
 **ISSN: 2575-9752**
 **[https://doi.org/10.25080/issn.2575-9752](https://doi.org/10.25080/issn.2575-9752)**
 
+**[SciPy 2023](https://conference.scipy.org/proceedings/scipy2023)**
+ *22nd Python in Science Conference - Austin, Texas (July 10—16, 2023)*
```
Comment (tkoyama010) [inline content/pages/proceedings.md:9]:
```suggestion
 *22nd Python in Science Conference - Austin, Texas (July 10–16, 2023)*
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

**Comment by achabotl** [comment]
Thanks for the review and reminding me to link the issue.

**Pull request merged**